[Tiew Wei Jian] iP

tiewweijian opened the pull request:
# DukeSoftware™️  🤖 
> Best companion for your daily hustle. | Made for [CS2103](https://nus-cs2103-ay2122s2.github.io/website/index.html)

***Are these things bothering you lately?*** 🤕 
- [x] Cannot keep track of what to do
- [x] Calendar too messy
- [x] Application not compatible with your computer

No worries, because our Duke Software provides you the ability to 
- Keep track of your daily tasks 👍🏻 
- Enter your daily tasks 👍🏻 
- Mark task as complete 👍🏻 

All you need to do, is to 
* Ensure you are using a compatible machine
* Download the program [here](https://nus-cs2103-ay2122s2.github.io/website/index.html)
* Give good reviews on Github

~The application cost $100000 on the Google Playstore.~ And the best part, this app is **free** for a limited time only. 

> This software is also built open source, so if you are a java programmer, you can check out our codes here! Here is a snippet of the `main` code.

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
### Never worry about your tasks ever again.

Commit:
Add Gradle support

tiewweijian committed:
Level-1

tiewweijian committed:
Level-2

tiewweijian committed:
Level-3

tiewweijian committed:
Level-4

tiewweijian committed:
Level-4

tiewweijian committed:
Level-5

tiewweijian committed:
A-TextUiTesting

tiewweijian committed:
Level-5

tiewweijian committed:
Level-6

tiewweijian committed:
A-Enums

tiewweijian committed:
Add Level-7 Files

tiewweijian committed:
Add date and time functionalities

tiewweijian committed:
Merge branch 'branch-Level-7'

tiewweijian committed:
Commit Level 8 improvements

tiewweijian committed:
Organise program into different objects

tiewweijian committed:
Move classes into packages

tiewweijian committed:
Add test case for tasks

tiewweijian committed:
Create jar file

tiewweijian committed:
Add JavaDoc

tiewweijian committed:
Correct coding standard

tiewweijian committed:
Create find function in program

tiewweijian committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

tiewweijian committed:
Merge branch-A-CodingStandard

tiewweijian committed:
Merge remote-tracking branch 'origin/branch-Level-9'

tiewweijian committed:
Update TaskList methods

tiewweijian committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

tiewweijian committed:
Add gradle support

tiewweijian committed:
Add CheckStyle checks

tiewweijian committed:
Merge remote-tracking branch 'origin/branch-A-CheckStyle'

domlimm commented on src/main/java/duke/ByeCommand.java:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Runs the command for user to exit the Duke-program.
+ */
+
```
This spacing could be removed.

domlimm commented on src/main/java/duke/ByeCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+public class ByeCommand extends Command {
+
+    /**
+     * Executes the command and exiting the program.
+     *
+     * @param taskList The TaskList of the current user.
+     * @param ui The user interface to show messages to users.
+     * @param storage The file system for reading and writing into the database.
+     */
+    @Override
```
Great job on documenting overridden methods! You followed the JavaDocs coding standard well!

- [x] Description of method
- [x] Spacing between description and other javadocs related
- [x] Not missing out params

😄

domlimm commented on src/main/java/duke/ByeCommand.java:
```diff
@@ -0,0 +1,23 @@
+        ui.outputMessage("Thank you for using Duke App. Have a nice day");
+        System.exit(1);
+    }
+
+
```
These break lines could be removed as well. In some other files too 👍🏻

domlimm commented on src/main/java/duke/ByeCommand.java:
```diff
@@ -0,0 +1,23 @@
+    @Override
+    void runCommand(TaskList taskList, Ui ui, Storage storage) {
+        ui.outputMessage("Thank you for using Duke App. Have a nice day");
+        System.exit(1);
```
Great 4 spaces indentation!

domlimm commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    /**
+     * This method returns a string of the type of task this is.
+     * @return a string that indicates what kind of task this is.
```
You could add a space to separate the description and other contents. Likewise your `getDescription` method too!

Noticed some other files require the same changes 👍🏻 

```suggestion
     * This method returns a string of the type of task this is.
     *
     * @return a string that indicates what kind of task this is.
```

domlimm commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
```
Great that you organised your imports lexicographically, all other files too!

domlimm commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+        return this.description
+                + " (by: " + formatter.format(date).toString() + ")";
+    }
+}
```
Missing a break line at the end of your file.

domlimm commented on src/main/java/duke/FindCommand.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    /**
+     *
```
You might have missed out this method's description for JavaDocs.

domlimm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
```
Well done on organising the imports properly!

domlimm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    /**
+     *
```
You might have missed out your method's description.

domlimm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+            }
+            return taskArrayList;
+        }
+    }
```
Great job on the indents! Everything follows the coding standard perfectly! 😄

domlimm commented on src/main/java/duke/TaskCommand.java:
```diff
@@ -0,0 +1,102 @@
+
+        ui.outputMessage(message);
+    }
+}
```
Again, followed the coding standard perfectly! Really pleasant to read!

domlimm commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,36 @@
+    @Test
+    public void testEventTask() throws IOException, DukeException, ParseException {
+        DateFormat formatter = new SimpleDateFormat("dd/MM/yy h:mm a");
+        assertEquals("hello (at: 10/10/21 10:50 PM)", new Event("hello", formatter.parse("10/10/2021 10:50 PM")).getDescription());
```
Do you think this line could be shortened? 😛 Likewise in `testDeadlineTask` too.

domlimm commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,36 @@
+import duke.*;
```
Should we import using `*`? 🤔

domlimm reviewed (commented):
Overall, I found it really pleasant to review your codes. There were really easy to read. I noted a few minor coding standard violations too, mainly `JavaDocs` related. Other than that, you followed the coding standard closely, and I had a really great time reviewing this PR! LGTMe, almost 🍻

bryans17 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+ */
+public class Deadline extends Task {
+
+    private Date date;
+    private DateFormat formatter = new SimpleDateFormat("dd/MM/yy h:mm a");
```
Good naming of variables. The names tell me all I need to know about what they are.

bryans17 commented on src/main/java/duke/FindCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+public class FindCommand extends Command {
+
+    private String taskToFind = null;
```
Initializing `taskToFind` to `null` might not be necessary, but it is nothing to worry about.

bryans17 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        String basePath = new File("").getAbsolutePath();
+        //file = new File( basePath + "/" + filePath);
```
Perhaps remove this commented-out line if you no longer require it.

bryans17 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+ */
+public class Storage {
+
+    private String filePath;
+    private File file;
+    private boolean isFileOpen = false;
```
Great naming of variables here. Follows the coding standard well.

bryans17 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+     * @param taskList the tasklist of the user in the current session
+     * @throws IOException when there is an error opening the file.
+     */
+    public void rewriteTask(TaskList taskList) throws IOException {
+        FileWriter rewriteWriter = new FileWriter(file, false);
+        for (Task task: taskList.getTaskArrayList()) {
+            String toPrint = task.toString() + task.getStatus() + task.getDescription();
+            rewriteWriter.write(toPrint + "\n");
+        }
+
+        rewriteWriter.close();
+    }
```
Would `saveTask` or `save` be a better and more intuitive name for this method?

bryans17 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,53 @@
+ * Represents a task that is created by an user.
+ */
+class Task {
+    enum Type {
+        E, T, D
+    }
```
Maybe consider giving these enums more descriptive names like `EVENT`, `TODO`, and `DEADLINE`, instead of `E`, `T`, `D`.

bryans17 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+     * @param taskName the search enquiry.
+     * @return an array list of tasks that contain the search enquiry.
+     */
+    public ArrayList<String> find(String taskName) {
+        String indentation = "    ";
+        int counter = 1;
+
+        ArrayList<Task> foundTask = new ArrayList<>();
+        for (Task task: taskArrayList) {
+            if (task.getDescription().toLowerCase(Locale.ROOT).contains(taskName.toLowerCase(Locale.ROOT))) {
+                foundTask.add(task);
+            }
+        }
+
+        ArrayList<String> stringOfTasks = new ArrayList<>();
+
+        if (foundTask.isEmpty()) {
+            stringOfTasks.add("No such task found");
+            return stringOfTasks;
+        }
+
+        stringOfTasks.add(indentation + "Here are the matching tasks in your list:");
+        for (Task task: foundTask) {
+            stringOfTasks.add(indentation + String.valueOf(counter) + ". "
+                    + task.toString() + task.getStatus() + " " + task.getDescription());
+            counter++;
+        }
+
+        foundTask.clear();
+        return stringOfTasks;
+    }
+}
```
Again, great job naming the variables! Found them to be very intuitive and descriptive.

bryans17 reviewed (commented):
Overall, well done on following the Java Coding Standard as well as having great code quality and descriptive variable names. Had a pleasant time reading your code and was very easy to understand what is going on. 👍🏻 
Keep up the good work. 😃

LeeEnEn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,66 @@
+     * @return an instance of a command for the system to run.
+     * @throws DukeException when the user types a command that is not recognised.
+     */
+    public Command processCommand(String command) throws DukeException {
+        if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.equals("bye")) {
+            return new ByeCommand();
+        } else if (command.contains("unmark")) {
+            int index = Integer.parseInt(command.split(" ")[1]);
+            return new MarkCommand(MarkCommand.MarkTypes.Unmark, index - 1);
+        } else if (command.contains("mark")) {
+            int index = Integer.parseInt(command.split(" ")[1]);
+            return new MarkCommand(MarkCommand.MarkTypes.Mark, index - 1);
+        } else if (command.contains("delete")) {
+            int index = Integer.parseInt(command.substring(7));
+            return new DeleteCommand(index - 1);
+        } else if (command.contains("todo")) {
+            return new TaskCommand(TaskCommand.TaskType.todo, command);
+        } else if (command.contains("event")) {
+            return new TaskCommand(TaskCommand.TaskType.event, command);
+        } else if (command.contains("deadline")) {
+            return new TaskCommand(TaskCommand.TaskType.deadline, command);
+        } else if (command.contains("/help")) {
+            return new HelpCommand();
+        } else if (command.contains("find")) {
+            return new FindCommand(command.substring(5));
+        } else {
+            throw new DukeException("Invalid Command.");
+        }
```
Perhaps you could use a switch statement to improve readability.

tiewweijian committed:
Add user interface for Duke

tiewweijian committed:
Update start point for project

Project now is starting from the Launcher class.

This class is redundant as it is possible to move the application start point to another class in the existing project.

Let's remove the Launcher class, and move the main method into Main class.

Using this will enable the project to be more streamlined, and the naming conventions more readable.

tiewweijian committed:
Create assertions at critical parts

Project currently does not have assertion points at various parts of the code. Gradle does not have assertions enabled.

At runtime, it is difficult to assert which part of the codes are causing
run time issue.

Let's create assertion points at various parts of the code. We will create assertion points at
* Storage - when data is unable to load up successfully.
* Main - when window is not able to load successfully.
* TaskList - when the new object created is not created successfully.
This will allow us to check critical portions of the code and understand which
part of the codes are causing runtime error.

tiewweijian committed:
Improve code quality and checkstyle violations

tiewweijian committed:
Merge pull request #1 from tiewweijian/branch-A-Assertions

Add assertions 
Add assertions into critical portions of the code. Assertions are included in the UI, and Storage.

tiewweijian committed:
Merge branch 'master' of https://github.com/tiewweijian/ip into branch-A-CodeQuality

tiewweijian committed:
Merge branch 'branch-A-CodeQuality'

tiewweijian committed:
Rectify code style violations

tiewweijian committed:
Merge branch 'branch-A-CodeQuality'

tiewweijian committed:
Set up continuous integration

tiewweijian committed:
Rectify checkstyle violations on test classes

tiewweijian committed:
Create priority tagging system.

Project currently now enables user to add tasks in chronological order.

However, different tasks has different level of priority despite having different due
dates.

Let's create a priority tagging system to each task. It is indicate with a $ in the data file,
and users can type $HIGH, $MEDIUM or $LOW to tag their task to the appropriate priority level.

snoidetx commented on src/main/java/duke/TaskCommand.java:
```diff
@@ -0,0 +1,124 @@
+     * @throws ParseException when there is an error in recognising the text.
+     */
+    @Override
+    String runCommand(TaskList taskList, Ui ui, Storage storage) throws IOException {
```
Hello! I am allocated to view your code in tutorial. I think this method is too long. Also, it is a bit deep nesting and forms an arrow-head style. You can consider abstracting out the lower-level methods, such as the `Parse()` function, because they are not in the same level as those higher-level methods like `addTask()`. That will make the code look more cleaner :)

tiewweijian committed:
Improve UI features

tiewweijian committed:
Resolve checkstyle issues

tiewweijian committed:
Update README.md

tiewweijian committed:
Update acknowledgements

tiewweijian committed:
Resolve coding standard issues

tiewweijian committed:
Create Launcher to rectify bugs

tiewweijian committed:
Fix bug for ByeCommand

tiewweijian committed:
Change help information
